Documentation/Cookbooks/translations.md

```diff
@@ -1,3 +1,260 @@
+
+```yaml
+parameters:
+    locale: fr
+````
+
+If you want you app to be available in multiple langagues, you'll have to use Symfony `{_locale}` parameter in your routes. Follow [the official routing documentation](https://symfony.com/doc/3.4/routing.html) to learn more.
+
+# Translations
+
+## Translating family name
+
+Here is an example of a simple family translation
+
+```yaml
+sidus_eav_model:
+    families:
+        Car: # <- this is the Family Code
+            data_class: MyCompany\MyBundle\Entity\Family\Car
+            attributeAsLabel: name
+            attributes:
+                name: ~
+                description: ~
+                color: ~
+        [...]            
+```
+
+ In the `messages.fr.yml` translation file, this results in
+ 
+ ```yaml
+ eav:
+     family:
+         Car: # <- this is the Family Code
+             label: Voiture
+         [...]
+ ```
+
+## Translating family attributes
+
+Let's say you want to translate your family's attribute
+```yaml
+sidus_eav_model:
+    families:
+        Car:
+            data_class: MyCompany\MyBundle\Entity\Family\Car
+            attributeAsLabel: name
+            attributes: # list of my car's attributes
+                name: ~
+                description: ~
+                color: ~
+        [...]            
+```
+
+In the `messages.fr.yml` translation file, this results in
+
+ ```yaml
+ eav:
+     family:
+         Car:
+             label: Voiture
+             attributes:
+                name:
+                    label: Nom
+                description:
+                    label: Description
+                color:
+                    label: Couleur
+         [...]
+ ```
+
+# Translating attributes
+
+You can always choose to translate attributes on a global level. In the `messages.fr.yml`, add an `attributes` key.
+
+> Translating attributes alsow impact filters in the datagrid
+
+```yaml
+eav:
+    family:
+    [...]
+    attribute:
+        name:
+            label: Nom
+        description:
+            label: Description
+        color:
+            label: Couleur    
+```
+
+Doing so result in translating all attributes named `descritpion` or `color` on the whole project, whatever family they are in.
+
+> Note : family translation overrides global translation.
+
+```yaml
+family:
+     Car:
+         label: Voiture
+         attributes:
+            description:
+                label: Description
+    [...]
+    attribute:
+        [...]
+        description:
+            label: Commentaire
+        [...]
+```
+
+In this case, `decription` will be translated as `Description` when it's related to the `Car` family but will be translated as `Commentaires` for every other families.
+
+## Translating base attributes
+
+Base attributes, such as `CreatedAt` or `UpdatedAt` can also be translated. In your translation file, simply add theses lines (at yaml root level, no indentation needed)
+
+ ```yaml
+"Created at": "Créé le"
+"Updated at": "Mis à jour le"
+```
```

Comment:
Also mention that for datagrids, you can translate any non-EAV attribute like this:
```yaml
datagrid:
    {datagrid.code}:
        {column.code}: Column label